### DISCUSSION TITLE
deepPartial with lazy

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [colinhacks/zod#1577] [Q&A]
So I have a bit of a conundrum with deepPartial. I did expect that it would make every property optional at least also those that might be attached schemas and also schemas that are lazy, but that does not seem the case, so if I have a parent that is deep partial and I lazy load a different schema which holds some required fields, that will not be deep partial and will throw.

**Comment by sschneider-ihre-pvs** [comment]
```ts
// something.dto.ts
export const SomethingSchema= (): z.ZodObject<
  Properties<ISomething>
> =>
  z.object<Properties<ISomething>>({
    id: z.string(),
    __typename: z.literal('Something').nullish(),
    somethingElse: z.lazy(() => SomethingElseSchema().nullish()),
  });

// somethingelse.dto.ts
export const SomethingElseSchema= (): z.ZodObject<
  Properties<ISomethingElse>
> =>
  z.object<Properties<ISomethingElse>>({
    id: z.string(),
    __typename: z.literal('SomethingElse').nullish(),
  });
  ```
if I then do something like `SomethingSchema().deepPartial()` I will still buy in the required field from `SomethingElseSchema` with doesn't become partial.

**Comment by sschneider-ihre-pvs** [comment]
@colinhacks or does it work as intended?

**Comment by NexZhu** [comment]
@colinhacks I have the same quetion.
